lib/transaction.js

```diff
@@ -26,6 +30,19 @@ var Transaction = module.exports = function(sequelize, options) {
   } else {
     this.id = this.name = Utils.generateUUID();
   }
+
+  // `transaction.promise` implementation #2805
+  // main propose is to allow chainable logic inside hooks
+  this.promise = new Promise(function () {});
```

Comment:
Wouldn't you prefer to call `promise._resolveCallback` or `this._rejectCallback`?
https://github.com/petkaantonov/bluebird/blob/master/src/promise.js#L467-L475